pkg/operator/ceph/controller.go

```diff
@@ -0,0 +1,173 @@
+	err = c.Watch(&source.Kind{
+		Type: &v1.ConfigMap{TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: v1.SchemeGroupVersion.String()}}}, &handler.EnqueueRequestForObject{}, predicateController(mgr.GetClient()))
+	if err != nil {
+		return err
+	}
+
+	// Watch for Secret (admission controller secret)
+	err = c.Watch(&source.Kind{
+		Type: &v1.Secret{TypeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: v1.SchemeGroupVersion.String()}}}, &handler.EnqueueRequestForObject{}, predicateController(mgr.GetClient()))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephClient object and makes changes based on the state read
+// and what is in the CephClient.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileConfig) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime logging interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileConfig) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the operator's configmap
+	opConfig := &v1.ConfigMap{}
+	logger.Debugf("reconciling %s", request.NamespacedName)
+	err := r.client.Get(context.TODO(), request.NamespacedName, opConfig)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("operator's configmap resource not found. will use default value or env var.")
+		} else {
+			// Error reading the object - requeue the request.
+			return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to get operator's configmap")
+		}
+	} else {
+		// Populate the operator's config
+		r.config.Parameters = opConfig.Data
+	}
+
+	// Reconcile Ceph CLI timeout, since the clusterd context is passed to by pointer to all CRD
+	// controllers they will receive the update
+	opcontroller.SetCephCommandsTimeout(r.config.Parameters)
+
+	// Reconcile Operator's logging level
+	err = reconcileOperatorLogLevel(opConfig.Data)
+	if err != nil {
+		return opcontroller.ImmediateRetryResult, err
+	}
+
+	// Reconcile discovery daemon
+	err = r.reconcileDiscoveryDaemon()
+	if err != nil {
+		return opcontroller.ImmediateRetryResult, err
+	}
+
+	// Reconcile webhook secret
+	// This is done in the predicate function
+
+	logger.Infof("%s done reconciling", controllerName)
+	return reconcile.Result{}, nil
+}
+
+func reconcileOperatorLogLevel(data map[string]string) error {
+	rookLogLevel := k8sutil.GetValue(data, "ROOK_LOG_LEVEL", "INFO")
+	logLevel, err := capnslog.ParseLevel(strings.ToUpper(rookLogLevel))
+	if err != nil {
+		return errors.Wrapf(err, "failed to load ROOK_LOG_LEVEL %q.", rookLogLevel)
```

Comment:
The reconcile is tiny in this controller so I think it does not hurt to retry, right?

Reply:
If they have a typo in the log level, it will be stuck retrying until they fix it. Either way they need to fix the typo, but seems like it is a small enough thing that we shouldn't fail the reconcile.